lib/apiservers/engine/backends/archive.go

```diff
@@ -164,38 +189,89 @@ func (c *Container) importToContainer(vc *viccontainer.VicContainer, target stri
 // ContainerStatPath stats the filesystem resource at the specified path in the
 // container identified by the given name.
 func (c *Container) ContainerStatPath(name string, path string) (stat *types.ContainerPathStat, err error) {
-	defer trace.End(trace.Begin(fmt.Sprintf("** statpath, name=%s, path=%s", name, path)))
+	defer trace.End(trace.Begin(name))
+	op := trace.NewOperation(context.Background(), "ContainerStatPath: %s", name)
 
-	fakeStat := &types.ContainerPathStat{}
-	fakeStat.Mode = os.ModeDir
+	op.Debugf("path received by statpath %s", path)
 
-	return fakeStat, nil
+	vc := cache.ContainerCache().GetContainer(name)
+	if vc == nil {
+		return nil, NotFoundError(name)
+	}
+
+	// trim / and . off from path and then append / to ensure the format is correct
+	path = filepath.Clean(path)
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+
+	mounts := mountsFromContainer(vc)
+	mounts = append(mounts, types.MountPoint{Destination: "/"})
+
+	// handle the special case of targeting a volume mount point before it exists.
+	// this will be important for non started container cp, will also be important
+	// to certain behaviors for diff on a non started container.
+	if stat, succeed := tryFakeStatPath(mounts, path); succeed {
+		op.Debugf("faking container stat path %#v", stat)
+		return stat, nil
+	}
+
+	primaryTarget := resolvePathWithMountPoints(op, mounts, path)
+	fs := primaryTarget.filterSpec
+
+	var deviceID string
+	var store string
+	if primaryTarget.mountPoint.Destination == "/" {
+		// Special case. / refers to container VMDK and not a volume vmdk.
+		deviceID = vc.ContainerID
+		store = constants.ContainerStoreName
+	} else {
+		deviceID = primaryTarget.mountPoint.Name
+		store = constants.VolumeStoreName
+	}
+
+	stat, err = c.containerProxy.StatPath(op, store, deviceID, fs)
+	if err != nil {
+		op.Errorf("error getting statpath: %s", err.Error())
+		switch err := err.(type) {
+		case *storage.StatPathNotFound:
+			return nil, ResourceNotFoundError(vc.Name, "file or directory")
+		default:
```

Comment:
Not sure what we can respond to the user since user has no knowledge of filterspec. We have the same problem for ImportArchive and ExportArchive. Do you suggest a retry of generating and encoding the filter spec or something like `error processing target`?